dag-syntax.cddl

```diff
@@ -0,0 +1,24 @@
+application = {
+    applicationName: tstr,
+    nodes: [ * node ],
+}
+
+node = {
+    task: tstr,
+    name: tstr,
+    type: "producer" / "prosumer" / "consumer",
+    description: tstr,
+    executor: "container",
+    container: {
+        image: tstr,
+        resources: {
+            cpu: uint,
+            memory: tstr, ; TODO: Discuss the unit we want to use here
```

Comment:
The Docker API expects the memory limit to be provided in bytes, see https://docs.docker.com/engine/api/v1.46/#tag/Container/operation/ContainerCreate